modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DefaultArchiveDeployer.java

```diff
@@ -152,7 +149,7 @@ private void undeployArtifactIfItsAPatch(T artifact) {
           }
           return false;
         }).findAny();
-    foundMatchingArtifact.ifPresent(deployer::undeploy);
+    foundMatchingArtifact.ifPresent(this::undeploy);
```

Comment:
I think that using this method instead of the original will cause the artifact to be fully undeployed (causing the data folder to be deleted), but that is not the expected behavior if when the artifact has to be undeployed because there is a new artifact patch

Reply:
No, it doesn't, that happens on uninstall I think. I tested  exactly that in the system tests. I had to use this method because if not the notifications were not send.

Reply:
That is incorrect. There are two methods to undeploy an artifact, undeploy and undeployArtifactWithoutUninstall. Undeploy method will uninstall the artifact on this line https://github.com/mulesoft/mule/blob/a29de774dff7a76800d99a6252863464642f6252/modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DefaultArchiveDeployer.java#L336